src/Core/Uri.re

```diff
@@ -49,20 +51,13 @@ module Scheme = {
       list(string)
       |> and_then(
            fun
-           | [scheme, ..._] =>
-             Internal.isKnownScheme(scheme)
-               ? succeed(scheme |> Internal.ofString)
-               : succeed(Custom(scheme))
+           | [scheme, ..._] => scheme |> ofString |> succeed
            | _ => fail("No scheme"),
```

Comment:
It's a good idea to be explicit and avoid the wildcard when at all possible so you get the benefit of exhaustiveness checking. And here you certainly can:

```suggestion
           | [scheme, ..._] => scheme |> ofString |> succeed
           | []=> fail("No scheme"),
```